pkg/condition/condition.go

```diff
@@ -0,0 +1,177 @@
+package condition
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"sync"
+
+	operators "github.com/operator-framework/api/pkg/operators/v2"
+	"github.com/pkg/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kubeWait "k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	patcher "github.com/openshift/windows-machine-config-operator/pkg/patch"
+	"github.com/openshift/windows-machine-config-operator/pkg/retry"
+)
+
+const (
+	upgradeableTrueReason   = "upgradeIsSafe"
+	upgradeableTrueMessage  = "The operator is safe for upgrade"
+	upgradeableFalseReason  = "upgradeIsNotSafe"
+	upgradeableFalseMessage = "The operator is currently processing sub-components. At least one controller is busy."
+	// OperatorConditionName is an environment variable set by OLM identifying the operator's OperatorCondition CR
+	OperatorConditionName = "OPERATOR_CONDITION_NAME"
+)
+
+var (
+	// ocName is the name of the OLM-managed OperatorCondition resource. Empty if operator is not OLM-managed
+	ocName string
+	// busyControllers is a representation of a set that holds the names of controllers that are currently reconciling
+	busyControllers map[string]struct{}
+	// resourceLock is a mutex lock to ensure thread-safe access of shared entities
```

Comment:
Please call out the entities this lock is for. Also given that this is protecting the map and the OperatorCondition it needs to be named appropriately.

Reply:
Having trouble keeping the name descriptive yet short. Going with `ocAndControllerSetLock`, LMK if you object/have other suggestions.